ShopApp/ShopApp/AppDelegate.swift

```diff
@@ -34,6 +35,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
     }
 
+    func application(_ application: UIApplication, open url: URL,
+                     options: [UIApplication.OpenURLOptionsKey: Any])
+      -> Bool {
```

Comment:
this line breakdown doesn't look too good. maybe line 40 can be moved to line 39 at least